File: client-norx/app/src/main/java/ua/in/meugen/myapplication/ui/activities/main/MainActivityModule.java
package ua.in.meugen.myapplication.ui.activities.main;

import dagger.Binds;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import ua.in.meugen.myapplication.app.di.scopes.PerActivity;
import ua.in.meugen.myapplication.app.di.scopes.PerFragment;
import ua.in.meugen.myapplication.ui.activities.base.BaseActivity;
import ua.in.meugen.myapplication.ui.activities.base.BaseActivityModule;
import ua.in.meugen.myapplication.ui.activities.main.fragment.MainFragment;
import ua.in.meugen.myapplication.ui.activities.main.fragment.MainFragmentModule;

@Module(includes = BaseActivityModule.class)
public abstract class MainActivityModule {

    @Binds @PerActivity
    abstract BaseActivity bindActivity(final MainActivity activity);

    @ContributesAndroidInjector(modules = MainFragmentModule.class)
    @PerFragment
    abstract MainFragment contributeMainFragment();
}

File: client-coroutines/app/src/main/java/ua/meugen/android/client/coroutines/ui/activities/main/MainActivityModule.kt
package ua.meugen.android.client.coroutines.ui.activities.main

import android.support.v7.app.AppCompatActivity
import dagger.Binds
import dagger.Module
import dagger.android.ContributesAndroidInjector
import ua.meugen.android.client.coroutines.app.di.PerActivity
import ua.meugen.android.client.coroutines.app.di.PerFragment
import ua.meugen.android.client.coroutines.ui.activities.base.BaseActivityModule
import ua.meugen.android.client.coroutines.ui.activities.main.fragment.MainFragment
import ua.meugen.android.client.coroutines.ui.activities.main.fragment.MainFragmentModule

@Module(includes = [BaseActivityModule::class])
abstract class MainActivityModule {

    @Binds @PerActivity
    abstract fun bindActivity(activity: MainActivity): AppCompatActivity

    @ContributesAndroidInjector(modules = [MainFragmentModule::class])
    @PerFragment
    abstract fun contributeMainFragment(): MainFragment
}